src/eth/transaction.rs

```diff
@@ -223,11 +280,209 @@ pub fn format_address(value: &Option<EthAddress>) -> BytesMut {
     }
 }
 
+/// Pads data with leading zeros to the specified length
+pub fn pad_leading_zeros(data: &[u8], length: usize) -> Vec<u8> {
+    if data.len() >= length {
+        return data.to_vec();
+    }
+    let mut zeros = vec![0; length - data.len()];
+    zeros.extend_from_slice(data);
+    zeros
+}
```

Comment:
This is sub-optimal for cases where you provide a collection of the desired length that you are okay with consuming. In this case, it will clone the collection.

Reply:
fixed